### PR TITLE
Kyma adapter

### PR DESCRIPTION
**Pull request by StanislavStefanov**
**Description**

Changes proposed in this pull request:
- add charts for kyma adapter component
- add kyma adapter skeleton

**Pull Request status**

- [x] Implementation
- [x] Unit tests
- [x] Integration tests
- [x] `chart/compass/values.yaml` is updated 
- [x] Mocks are regenerated, using the automated script

### CODE REVIEW

**Comment by StanislavStefanov** [comment]
reset-vm

File: components/kyma-adapter/internal/claims-validator/validate.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	"context"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/apperrors"
+	"github.com/kyma-incubator/compass/components/director/pkg/idtokenclaims"
+	"github.com/kyma-incubator/compass/components/hydrator/pkg/tenantmapping"
+	"github.com/pkg/errors"
+)
+
+// ClaimsValidator implements the ClaimsValidator interface
+type ClaimsValidator struct{}
```
Comment (PetarTodorovv) [inline components/kyma-adapter/internal/claims-validator/validate.go:13]: Check the comment for the package/import/structure renaming

File: components/kyma-adapter/internal/handler/handler.go [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+import (
+	"net/http"
+
+	"github.com/machinebox/graphql"
+)
+
+// AdapterHandler processes received requests
+type AdapterHandler struct {
+	GqlClient *graphql.Client
```
Comment (PetarTodorovv) [inline components/kyma-adapter/internal/handler/handler.go:11]: `directorGQLClient`? or something similar?

**Review by PetarTodorovv — CHANGES_REQUESTED** [review]

File: components/director/pkg/auth-middleware/middleware.go [hunk 3]
```diff
@@ -95,50 +97,35 @@ func (a *Authenticator) Handler() func(next http.Handler) http.Handler {
...
+			tokenClaims, statusCode, err := a.processToken(ctx, r)
+			if err != nil {
+				apperrors.WriteAppError(ctx, w, err, statusCode)
+				return
 			}
 
-			if scenarioGroupsValue := r.Header.Get(ctxScenarioGroupsKey); scenarioGroupsValue != "" {
-				log.C(ctx).Infof("Found %s header in request with value: %s", ctxScenarioGroupsKey, scenarioGroupsValue)
-				groups := strings.Split(strings.ToUpper(scenarioGroupsValue), ",")
+			ctx = tokenClaims.ContextWithClaimsAndProviderTenant(ctx)
```
[the remark this answers is not in the document]
Comment (StanislavStefanov) [reply components/director/pkg/auth-middleware/middleware.go:126]: For this flow the provider and consumer tenant should be the same. The cert_service context provider should be matched which will provide the providerTenant. No other context providers are expected to match in the productive scenarios. When the tenant object map does not contain entries for consumer they will be created with the same values as the entries for the provider. As the two values are equal we can use either of them.

In the local scenarios the certificates used in the tests are whitelisted which causes second object context provider to be matched. The second object context provider provides empty values for the consumer tenant. As there are entries in the map the provider values are not duplicated in the consumer values regardless of them being empty. In that case the consumer tenant that reaches us in the context is empty and the request validation fails. The provider tenant on the other side is correctly populated.

For both productive and local setup the provider tenant will do the work.

File: chart/compass/charts/kyma-adapter/values.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+deployment:
+  minReplicas: 1
+  maxReplicas: 2
```
[the remark this answers is not in the document]
Comment (StanislavStefanov) [reply chart/compass/charts/kyma-adapter/values.yaml:3]: I am not sure if there is need for that. The only reason I added the configuration is that if the min and max replicas are equal we will be receiving alerts for running at max replicas

File: chart/compass/charts/kyma-adapter/templates/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+            - name: APP_TENANT_INFO_ENDPOINT
+              value: https://{{ .Values.global.gateway.tls.host }}.{{ .Values.global.ingress.domainName }}{{ .Values.global.director.info.path }}
+            - name: APP_TENANT_INFO_REQUEST_TIMEOUT
+              value: {{ .Values.global.kymaAdapter.tenantInfo.requestTimeout }}
+            - name: APP_TENANT_INFO_INSECURE_SKIP_VERIFY
+              value: {{ $.Values.global.http.client.skipSSLValidation | quote }}
+          livenessProbe:
+            httpGet:
+              port: {{.Values.deployment.args.containerPort }}
+              path: "healthz"
```
[the remark this answers is not in the document]
Comment (StanislavStefanov) [reply chart/compass/charts/kyma-adapter/templates/deployment.yaml:80]: The healthz endpoint is on the main router and the productive handler is on a subrouter with path prefix kyma-adapter.

File: chart/compass/charts/kyma-adapter/values.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (PetarTodorovv) [reply chart/compass/charts/kyma-adapter/values.yaml:3]: hm, maybe it's okay to have productive overrides - min/max replicas due to high availability, also resources limits/request, istio resources overrides?

**Review by PetarTodorovv — APPROVED** [review]

**Review by dzahariev — DISMISSED** [review]

**Review by dzahariev — APPROVED** [review]

**Review by PetarTodorovv — APPROVED** [review]

**Pull request merged**